thorlcr/graph/thgraphslave.cpp

```diff
@@ -2107,11 +2121,16 @@ class CFileCache : public CSimpleInterfaceOf<IThorFileCache>
              */
         }
         file->setActivity(&activity); // an activity needed by IActivityReplicatedFile, mainly for logging purposes.
```

Comment:
Can this cause any problems being modified with something else using it?  (E.g. if this activity then stops using it and is destroyed, but the original activity continues running.)

Reply:
No, this isn't new btw, it's also not particularly nice, but it's only used for tracing, when and if the file is closed by the cache limiting and reopened. This set ensures it will have the current activity (essentially the current context), when it is looked up, before it gets there.